src/dmd/cppmangle.d

```diff
@@ -101,8 +101,52 @@ bool isPrimaryDtor(const Dsymbol sym)
     return dtor == ad.primaryDtor;
 }
 
+/// Context used when processing pre-semantic AST
+private struct Context
+{
+    /// Template instance of the function being mangled
+    TemplateInstance ti;
+    /// Function declaration we're mangling
+    FuncDeclaration fd;
+    /// Current type / expression being processed (semantically analyzed)
+    RootObject res;
+
+    @disable ref Context opAssign(ref Context other);
+    @disable ref Context opAssign(Context other);
+
+    /**
+     * Helper function to track `res`
+     *
+     * Params:
+     *   next = Value to set `this.res` to
+     *          If `this.res` is `null`, the expression is not evalutated
+     *          This allow this code to be used even when no context is needed.
```

Comment:
As the first line didn't have a period, when reading this I started the parsing the second line as part of the same sentence.

Not disputing whether it looks fine in the generated docs, however I won't likely ever read from the web. :-)